Test suite is failing and blocking PRs

This is for discussion of the problem that is making all PRs suddenly start failing.

The related error in Travis (and on my local machine) is:

```
> papaparse@4.6.0 test-phantomjs /home/travis/build/mholt/PapaParse
> node tests/test.js --phantomjs
Error loading resource https://www.chaijs.com/chai.js (6). Details: SSL handshake failed
TypeError: 'undefined' is not an object (evaluating 'chai.assert')
   at http://localhost:8071/tests/test-cases.js:8
npm ERR! Linux 4.4.0-101-generic
npm ERR! argv "/home/travis/.nvm/versions/node/v6.14.4/bin/node" "/home/travis/.nvm/versions/node/v6.14.4/bin/npm" "run" "test-phantomjs"
npm ERR! node v6.14.4
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! papaparse@4.6.0 test-phantomjs: `node tests/test.js --phantomjs`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the papaparse@4.6.0 test-phantomjs script 'node tests/test.js --phantomjs'.
```

A possible observation of note is that the url it is trying to get chai from is `https://www.chaijs.com/chai.js` but [where we import it](https://github.com/mholt/PapaParse/blob/master/tests/tests.html#L9) we ask for `http://chaijs.com/chai.js` (note one is http and the other is https, and the error is related to SSL).

--- a/package.json
+++ b/package.json
@@ -36,7 +36,7 @@
 	"license": "MIT",
 	"main": "papaparse.js",
 	"devDependencies": {
-		"chai": "^4.1.2",
+		"chai": "^4.2.0",
 		"connect": "^3.3.3",
 		"eslint": "^4.19.1",
 		"grunt": "^1.0.2",

--- a/tests/tests.html
+++ b/tests/tests.html
@@ -5,8 +5,8 @@
 		<script src="../node_modules/mocha/mocha.js"></script>
 		<link rel="stylesheet" href="../node_modules/mocha/mocha.css" />
 		<script src="../papaparse.js"></script>
-
-		<script src="http://chaijs.com/chai.js"></script>
+		
+		<script src="../node_modules/chai/chai.js"></script>
 
 		<script>mocha.setup('bdd')</script>
 		<script src="test-cases.js"></script>

